### ISSUE TITLE
"Change inst." in palette works for range selection but Add>Text>Instrument change does not

### ISSUE THREAD

**Issue by zacjansheski** [musescore/MuseScore#21222]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

If I make a range selection, I am unable to add an instrument change from the top menu via Add>Text>Instrument

I am able to do so from the palette, these should behave the same.

### Steps to reproduce

1. Make a range selection
2. Add>Text>Instrument

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/69917893/85fb9a07-c3d3-49fa-9398-b31cce58093a

### MuseScore Version

Master

### Regression

No.

### Operating system

All

### Additional context

Same since MS3

**Comment by zacjansheski** [comment]
I forgot to demo in the video, but when doing a list selection,
Add>Text>Instrument only adds the instrtument change to the last selection
Adding from the palette adds to every selection

**Comment by bkunda** [comment]
The problem here is actually that it's even possible to add an instrument change to a range selection (of multiple staves). The UX for this is appalling: the Instrument dialog just keeps re-appearing until you've moved through all the staves, and there is no feedback to the user about what is going on. 

I would suggest that the behaviour for the palette item "Change instr." be aligned with the current behaviour of the menu item Add > Text > Instrument change. 

@mathesoncalum FYI

**Comment by cbjeukendrup** [comment]
> I would suggest that the behaviour for the palette item "Change instr." be aligned with the current behaviour of the menu item Add > Text > Instrument change.

But perhaps the UX of that "No note or rest selected" dialog is not ideal either... as an alternative, would it be an idea to apply the element just to the topmost staff only in case of a range selection?

**Comment by bkunda** [comment]
> as an alternative, would it be an idea to apply the element just to the topmost staff only in case of a range selection?

Yes, this would be fine to I suppose. It'd still be clear to the user that the "change instrument" is only taking effect on the uppermost stave (once the text is inserted). 

The "no note or rest selected" dialog is certainly not as useful as clearer messaging along the lines of "instrument changes can only be applied to a single stave at a time". But your suggestion of simply adding the text to the uppermost stave at least doesn't punish the user with no action and an error dialog.

**Comment by mathesoncalum** [comment]
Ty for the info @bkunda @cbjeukendrup - I'll be taking a look at this soon.

Any thoughts on adding instrument changes to list selections? Currently adding from the palette suffers from the same issue as range selections where the dialog will continually reappear with no feedback to the user. Adding from add->text->instrument change only applies the change to the last element added to the selection which also feels a bit odd.

https://github.com/musescore/MuseScore/assets/47119327/f278a6b7-dddc-4a01-9c1b-22888b2c0e75
